### PR TITLE
Conda cpp post build checks

### PR DESCRIPTION
**Pull request by robertmaynard**
Introduce the concept of post C++ build checks. Includes the first optional check which scans a the produced C++ conda libraries for any weak CUDA kernels.

### CODE REVIEW

**Comment by robertmaynard** [comment]
Verification runs of this workflow on raft:
Job passes as expected when given the correct symbol exclusion list: https://github.com/rapidsai/raft/actions/runs/6697280824/job/18198270374

Job fails as expected when the default symbol exclusion list is used:
https://github.com/rapidsai/raft/actions/runs/6699928979/job/18205855860?pr=1935

File: .github/workflows/conda-cpp-post-build-checks.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  repository-projects: none
+  security-events: none
+  statuses: none
+
+jobs:
+  check-symbols:
+    if: ${{ inputs.enable_check_symbols }}
+    runs-on: linux-amd64-cpu4
+    container:
+      image: rapidsai/ci-wheel:cuda12.0.1-centos7-py3.10
```
Comment (ajschmidt8) [inline .github/workflows/conda-cpp-post-build-checks.yaml:47]:
```suggestion
      image: rapidsai/ci-wheel:latest
```

Does the version of this container matter?

If not, can we switch it to `latest`?

The versions used for our `latest` tags are specified in the `rapidsai/ci-imgs` repository here: https://github.com/rapidsai/ci-imgs/blob/main/latest.yaml

**Review by ajschmidt8 — COMMENTED** [review]
Just one question

File: .github/workflows/conda-cpp-post-build-checks.yaml (hunk 1 above)
Comment (robertmaynard) [reply .github/workflows/conda-cpp-post-build-checks.yaml:47]: The version doesn't matter. So moving to use `latest`

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**